api/crl/crl_test.go

```diff
@@ -1,86 +1,55 @@
 package crl
 
 import (
-	"bytes"
+	"crypto/x509"
+	"encoding/base64"
 	"encoding/json"
-	"github.com/cloudflare/cfssl/api"
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
 	"testing"
+	"time"
+
+	"github.com/cloudflare/cfssl/api"
+	"github.com/cloudflare/cfssl/certdb"
+	"github.com/cloudflare/cfssl/certdb/sql"
+	"github.com/cloudflare/cfssl/certdb/testdb"
 )
 
 const (
-	cert       = "../../crl/testdata/caTwo.pem"
-	key        = "../../crl/testdata/ca-keyTwo.pem"
-	serialList = "../../crl/testdata/serialList"
-	expiryTime = "2000"
+	fakeAKI       = "fake aki"
+	testCaFile    = "../testdata/ca.pem"
+	testCaKeyFile = "../testdata/ca_key.pem"
 )
 
-type testJSON struct {
-	Certificate        string
-	SerialNumber       []string
-	PrivateKey         string
-	ExpiryTime         string
-	ExpectedHTTPStatus int
-	ExpectedSuccess    bool
-}
-
-var tester = testJSON{
-	Certificate:        cert,
-	SerialNumber:       []string{"1", "2", "3"},
-	PrivateKey:         key,
-	ExpiryTime:         "2000",
-	ExpectedHTTPStatus: 200,
-	ExpectedSuccess:    true,
-}
-
-func newTestHandler(t *testing.T) http.Handler {
-	return NewHandler()
-}
-
-func TestNewHandler(t *testing.T) {
-	newTestHandler(t)
-}
-
-func newCRLServer(t *testing.T) *httptest.Server {
-	ts := httptest.NewServer(newTestHandler(t))
-	return ts
-}
-
-func testCRLCreation(t *testing.T, issuingKey, certFile string, expiry string, serialList []string) (resp *http.Response, body []byte) {
-
-	ts := newCRLServer(t)
-	defer ts.Close()
-
-	obj := map[string]interface{}{}
-
-	if certFile != "" {
-		c, err := ioutil.ReadFile(certFile)
-		if err != nil {
-			t.Fatal(err)
-		}
-		obj["certificate"] = string(c)
+func prepDB() (certdb.Accessor, error) {
+	db := testdb.SQLiteDB("../../certdb/testdb/certstore_development.db")
+	expirationTime := time.Now().AddDate(1, 0, 0)
+	var cert = certdb.CertificateRecord{
+		Serial:    "1",
+		AKI:       fakeAKI,
+		Expiry:    expirationTime,
+		PEM:       "revoked cert",
+		Status:    "revoked",
+		RevokedAt: time.Now(),
+		Reason:    4,
 	}
 
-	obj["serialNumber"] = serialList
-
-	if issuingKey != "" {
-		c, err := ioutil.ReadFile(issuingKey)
-		if err != nil {
-			t.Fatal(err)
-		}
-		obj["issuingKey"] = string(c)
+	dbAccessor := sql.NewAccessor(db)
+	err := dbAccessor.InsertCertificate(cert)
+	if err != nil {
+		return nil, err
 	}
 
-	obj["expireTime"] = expiry
+	return dbAccessor, nil
+}
 
-	blob, err := json.Marshal(obj)
-	if err != nil {
-		t.Fatal(err)
-	}
+func testGetCrl(t *testing.T, dbAccessor certdb.Accessor) (resp *http.Response, body []byte) {
+	handler, _ := NewHandler(dbAccessor, testCaFile, testCaKeyFile)
```

Comment:
In tests, you should always check the error to make sure your expectations are correct and to catch failures to match these expectations if the code changes later.